### PR TITLE
Remove Probes from Disallowed fields in EventListener

### PR DESCRIPTION
**Pull request by khrm**
We can set liveness, readiness and start probe now in EventListener.

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/triggers/v1beta1/event_listener_validation.go [hunk 1]
```diff
@@ -232,9 +232,6 @@ func containerFieldMask(in *corev1.Container) *corev1.Container {
 	out.Image = ""
 	out.Args = nil
 	out.Ports = nil
-	out.LivenessProbe = nil
-	out.ReadinessProbe = nil
-	out.StartupProbe = nil
```
Comment (savitaashture) [inline pkg/apis/triggers/v1beta1/event_listener_validation.go:237]: this will break the scenario for customresource

Comment (khrm) [reply pkg/apis/triggers/v1beta1/event_listener_validation.go:237]: Why?

Comment (savitaashture) [reply pkg/apis/triggers/v1beta1/event_listener_validation.go:237]: Added all changes in this commit https://github.com/tektoncd/triggers/pull/1620/commits/fed390f91ad29bb77e027fb3832f7bdadbeb6ec9

**Comment by savitaashture** [comment]
@khrm @dibyom PR is ready to review

I have added unit test and updated doc as well

PTAL
Thank you

**Comment by khrm** [comment]
@savitaashture Squash all three commit into one and I will give lgtm.

**Comment by khrm** [comment]
Actually, only the first two commits need to be squashed. The last one deals with something else. So it's okay I think.

**Comment by khrm** [comment]
Or open another pr and I will give lgtm.

**Pull request closed without merging**